### ISSUE TITLE
Implement Sender

### ISSUE THREAD

**Issue by fraguada** [fraguada/SpeckleRhino#27]

**Comment by fraguada** [comment]
The sender logic, as discussed:

- Once a sender is created, a corresponding parent layer is added to the Rhino Document LayerTable.
- The sender listens to changes on this layer.  If a child layer is added/removed, renaming child layer, reordering, adding/removing objects to child layers, etc.
- Changes are pushed

**Comment by didimitrie** [comment]
Main future ux question: what do we do with objects on parent layer?

**Comment by fraguada** [comment]
Yeah, I was thinking of that.  Somehow we have to forbid adding stuff to the parent layer.  We'll have similar issues for changing the name of the parent layer.  

For stuff that is added to the parent layer, we can listen for it, and pass the objects to a 'default' or something child layer.

For changing the name of the parent layer, perhaps we only allow changing the name from the Speckle Panel UI? If a user tries to change it in Rhino, it gets forced back to whatever it is in the stream. So only child layer changes are allowed on the Rhino layers panel, and visibility...thats kind of weird though...
